network/src/testutils/test_node.rs

```diff
@@ -345,3 +365,82 @@ pub fn mock_conn_metadata(
     }
     metadata
 }
+
+/// Creates a mock connection based on the `Validator` to `Validator` connection
+pub fn validator_mock_connection(
+    origin: ConnectionOrigin,
+    protocol_ids: &[ProtocolId],
+) -> (PeerNetworkId, ConnectionMetadata) {
+    let peer = PeerNetworkId::new(NetworkId::Validator, PeerId::random());
+    let metadata = mock_conn_metadata(peer, PeerRole::Validator, origin, Some(protocol_ids));
+    (peer, metadata)
+}
+
+/// Creates a mock connection based on the `Vfn` to `Validator` connection
+pub fn vfn_validator_mock_connection(
+    origin: ConnectionOrigin,
+    protocol_ids: &[ProtocolId],
+) -> (PeerNetworkId, ConnectionMetadata) {
+    let peer = PeerNetworkId::new(NetworkId::Vfn, PeerId::random());
+    let metadata = match origin {
```

Comment:
nit: you could just `match origin` to extract the peer role, right? Everything else is the same? e.g.,

```
let role = match origin {
...
};
let metadata = mock_conn_metadata(peer, role, origin, Some(protocol_ids));
```

Reply:
Yeah, I can fix that.